Backend/Tournament_Public/controllers/publicUserController.js

```diff
@@ -0,0 +1,151 @@
+
+
+
+/** Controller to get all tournaments
+ * @async
+ * @param {Object} req - The request object.
+ * @param {Object} res - The response object.
+ * @param {Function} next - The next middleware function.
+ * @returns {Json} Return Upcoming tournament based on upcoming dates.
+ * @throws {error} If input invalid or missing fields.
+ */
+
+exports.getAllTournaments = async (req, res, next) => {
+    try {
+        const allTournaments = await TournamentUserService.getAllTournaments();
+        if (!allTournaments || allTournaments.length === 0) {
+            const error = new Error('No tournaments retrieved');
+            error.statusCode = 404;
+            return next(error);
+        }
+        res.status(200).json({
+            "success": true,
+            "status" : 200,
+            "message": "successfully retrieved all tournaments",
+            "content": allTournaments
+        });
+    } catch (err) {
+        console.error('Error retrieving tournaments:', err);
+        next(err);  // Pass the error to the middleware
+    }
+};
+
+/** Controller to get upcoming tournaments
+ * @async
+ * @param {Object} req - The request object.
+ * @param {Object} res - The response object.
+ * @param {Function} next - The next middleware function.
+ * @returns {Json} Return Upcoming tournament based on upcoming dates.
+ * @throws {error} If input invalid or missing fields.
+ */
+
+exports.getUpcomingTournaments = async (req, res, next) => {
+    try {
+        const upcomingTournaments = await TournamentUserService.getUpcomingTournaments();
+        if (!upcomingTournaments || upcomingTournaments.length === 0) {
+            const error = new Error('No upcoming tournaments');
+            error.statusCode = 404;
+            return next(error);
+        }
+        res.status(200).json({
+            "success": true,
+            "status" : 200,
+            "message": "successfully retrieved upcoming tournaments",
+            "content": upcomingTournaments
+        });
+    } catch (err) {
+        console.error('Error retrieving upcoming tournaments:', err);
+        next(err);  // Pass the error to the middleware
+    }
+};
+
+/**  Controller to get in-progress tournaments
+ * @async
+ * @param {Object} req - The request object.
+ * @param {Object} res - The response object.
+ * @param {Function} next - The next middleware function.
+ * @returns {Json} Return incoming tournament based on upcoming dates.
+ * @throws {error} If input invalid or missing fields.
+ */
+
+exports.getInProgressTournaments = async (req, res, next) => {
+    try {
+        const inProgressTournaments = await TournamentUserService.getInProgressTournaments();
+        if (!inProgressTournaments || inProgressTournaments.length === 0) {
+            const error = new Error('No in-progress tournaments');
+            error.statusCode = 404;
+            return next(error);
+        }
+        res.status(200).json({
```

Comment:
Not sure if i seeing wrong? but i think you only middleware for error and not success.